### PR TITLE
Add support for the MINIDUMP_HANDLE_DATA_STREAM stream

### PR DESCRIPTION
**Pull request by gabrielesvelto**
This covers part of issue #324

### CODE REVIEW

**Comment by gabrielesvelto** [comment]
Tests are missing, I'll add them next week

**Comment by gabrielesvelto** [comment]
I've manually tested this on Windows-generated minidumps that had the stream populated

**Comment by gabrielesvelto** [comment]
@Gankra I've tested this with actual minidumps and everything seems to be working fine, but it seems like I messed up something in *minidump-synth*. Somehow the test fail with the value of `handle` being returned in the wrong endianness, so instead of `0x000000000000007b` I get `0x7b00000000000000`. I can't figure out what I'm doing wrong though.

File: minidump-synth/src/lib.rs [hunk 1]
```diff
@@ -758,7 +782,8 @@ impl<T: ListItem> ExList<T> {
         let section = Section::with_endian(endian)
             .D32(12)
```
Comment (Gankra) [inline minidump-synth/src/lib.rs:783]:
```suggestion
            .D32(12 + padding as u32)
```

**Review by Gankra — APPROVED** [review]

File: minidump-common/src/format.rs [hunk 2]
```diff
@@ -2459,3 +2462,86 @@ bitflags! {
...
+    MiniEventInformation1,
+    MiniSectionInformation1,
+    MiniSemaphoreInformation1,
+    MiniHandleObjectInformationTypeMax,
+}
+
+// OS-specific handle object information. Microsoft headers state that it can
+// change based on the target platform. The object is larger than this structure
+// (as specified by `size_of_info`) and the remaining data is stored after the
+// `size_of_info` field. The format of this information is not specified.
```
[the remark this answers is not in the document]
Comment (gabrielesvelto) [reply minidump-common/src/format.rs]: Fixed and added some extra info to the structure itself

File: minidump-synth/src/lib.rs (hunk 1 above)
Comment (gabrielesvelto) [reply minidump-synth/src/lib.rs]: I went with passing the entire header size and then figuring out the padding from there, it's harder to get it wrong then

Comment (gabrielesvelto) [reply minidump-synth/src/lib.rs]: Thanks a lot for spotting what was wrong with this code

**Pull request merged**